augur/frequency_estimators.py

```diff
@@ -54,18 +59,29 @@ def get_pivots(observations, pivot_interval, start_date=None, end_date=None, piv
     pivot_end = end_date if end_date else np.ceil(np.max(observations) / pivot_frequency) * pivot_frequency
 
     if pivot_interval_units == "months":
-        offset = "%sMS" % pivot_interval
+        delta_keywords = {"months": pivot_interval}
     elif pivot_interval_units == "weeks":
-        offset = "%sW" % pivot_interval
+        delta_keywords = {"weeks": pivot_interval}
     else:
         raise ValueError(f"The given interval unit '{pivot_interval_units}' is not supported.")
 
-    datetime_pivots = pd.date_range(
-        float_to_datestring(pivot_start),
-        float_to_datestring(pivot_end),
-        freq = offset
-    )
-    pivots = np.array([timestamp_to_float(pivot) for pivot in datetime_pivots])
+    # Construct standard Python date instances from numeric dates via ISO-8601
+    # dates and the corresponding delta time for the interval between pivots.
+    start = datetime.datetime.strptime(float_to_datestring(pivot_start), "%Y-%m-%d")
+    end = datetime.datetime.strptime(float_to_datestring(pivot_end), "%Y-%m-%d")
+    delta = relativedelta(**delta_keywords)
+
+    # Start calculating pivots from the end date (inclusive), working backwards
+    # in time by a time delta that matches the user-requested interval. Include
+    # the start date in the final pivots when the interval spacing allows that
+    # date to be included.
+    pivots = deque([])
+    pivot = end
+    while pivot >= start:
+        pivots.appendleft(pivot)
```

Comment:
_FYI, non-blocking_

```suggestion
    pivots = []
    pivot = end
    while pivot >= start:
        pivots.insert(0, pivot)
```

This accomplishes the same thing without using an extra import.

Reply:
Fair point, although I don't mind the slight extra import cost for the slight readability improvement of working with the deque interface.